packages/design-system/src/components/number-field/types.ts

```diff
@@ -0,0 +1,53 @@
+import type {
+  NumberFieldProps as AriaNumberFieldProps,
+  NumberFieldRenderProps as RACNumberFieldRenderProps,
+} from 'react-aria-components';
+import type { PartialDeep } from 'type-fest';
+import type {
+  ButtonClassNames,
+  ButtonProps,
+  InputClassNames,
+} from '../../components';
+import type {
+  AsType,
+  OmitProtectedProps,
+  RenderPropsChildren,
+} from '../../types';
+
+export type NumberFieldClassNames = PartialDeep<{
+  container: string;
+  decrement: ButtonClassNames;
+  description: string;
+  error: string;
+  group: string;
+  increment: ButtonClassNames;
+  input: InputClassNames;
+  label: string;
+  numberField: string;
+}>;
+
+export type NumberFieldSizes = 'sm' | 'lg';
+
+export type NumberFieldMapping = {
+  description: Partial<Record<NumberFieldSizes, string>>;
+  error: Partial<Record<NumberFieldSizes, string>>;
+  increment: Partial<Record<NumberFieldSizes, OmitProtectedProps<ButtonProps>>>;
+  decrement: Partial<Record<NumberFieldSizes, OmitProtectedProps<ButtonProps>>>;
+};
+
+type BaseNumberFieldProps = {
+  children?: RenderPropsChildren<NumberFieldRenderProps>;
```

Comment:
You don't need to redefine `children` unless you're changing the type. This type is the same as the children type from the RAC